frontend/src/components/pages/profile/Profile.tsx

```diff
@@ -1,10 +1,186 @@
+    setPerson({...person, pronouns: event.target.value});
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    (async() => {
+      const success = (person.email === "" || person.nickname === "")
+        ? false /** At least one required field is missing */
+        : await profileHandlerService.updatePerson(person);
+      setSubmitSuccess(success);
+    })();
+  };
 
-export const Profile = () => {
   return (
-    <div>
-      <h2>Profile</h2>
-      <p>This is the Profile page.</p>
+    <div className={classes.root}>
+      <form>
+        <div>
+          <TextField
+            fullWidth
+            helperText="to be displayed around CoffeeHouse"
+            id="nickname"
+            InputLabelProps={{
+              shrink: true,
+            }}
+            label="Nickname"
+            margin="normal"
+            onChange={handleNameChange}
+            placeholder="Nickname"
+            style={{ margin: 8 }}
+            value={person? person.nickname : ""}
+            variant="outlined"
+          />
+          <TextField
+            fullWidth
+            helperText="to be displayed around CoffeeHouse"
+            id="pronouns"
+            InputLabelProps={{
+              shrink: true,
+            }}
+            label="Pronouns"
+            margin="normal"
+            onChange={handlePronounsChange}
+            placeholder="Pronouns"
+            style={{ margin: 8 }}
+            value={person? person.pronouns : ""}
+            variant="outlined"
+          />
+          <TextField
+            fullWidth
+            id="email"
+            InputLabelProps={{
+              shrink: true,
+            }}
+            label="Email"
+            margin="normal"
+            onChange={handleEmailChange}
+            placeholder="Email"
+            style={{ margin: 8 }}
+            value={person? person.email : ""}      
+            variant="outlined"
+          />
+        </div>
+        <div>
+          <Button 
+            color="primary" 
+            onClick={handleSubmit}
+            style={{ margin: 8 }} 
+            variant="contained" 
+          >
+            Submit
+          </Button>
+        </div>
+        <DisplaySubmitStatus success={submitSuccess} />
+      </form>
     </div>
   );
 }
+
+/**
+ * Displays a success or failure message to the user after they submit a 
+ * a profile change. 
+ */
+function DisplaySubmitStatus(props) {
+  const success = props.success;
+
+  if (success === undefined) {
+    return (<div></div>);
+  } else {
+    if (success) {
+      return (
+        <div>
+          <p style={{ color: 'green', margin: 8 }}>
+            Profile successfully updated.
+          </p>
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <p style={{ color: 'red', margin: 8 }}>
+            Something went wrong.
+          </p>
+        </div>
+      )
```

Comment:
```suggestion
      );
```
Please add semicolon at end of line.